lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -213,20 +215,28 @@ public function fetch($fetchMode = null)
                     $ctorArgs  = isset($args[2]) ? $args[2] : array();
                 }
 
-                $result = db2_fetch_object($this->_stmt);
+                $result = @db2_fetch_object($this->_stmt);
 
                 if ($result instanceof \stdClass) {
                     $result = $this->castObject($result, $className, $ctorArgs);
                 }
-
-                return $result;
+                break;
             case \PDO::FETCH_NUM:
-                return db2_fetch_array($this->_stmt);
+                $result = @db2_fetch_array($this->_stmt);
+                break;
             case \PDO::FETCH_OBJ:
-                return db2_fetch_object($this->_stmt);
+                $result = @db2_fetch_object($this->_stmt);
+                break;
             default:
                 throw new DB2Exception("Given Fetch-Style " . $fetchMode . " is not supported.");
         }
+
+        $error = db2_stmt_errormsg();
```

Comment:
Actually that won't work because we can't mock private functions. @Ocramius how would you suggest testing this? In my company's projects we use [php-mock/php-mock-phpunit](https://github.com/php-mock/php-mock-phpunit) to mock built-in functions, but Doctrine doesn't use this. Otherwise, could create a `DB2StatementMock` subclass which mocks `doFetch()`, and make that the SUT, but then we're not really testing `DB2Statement` directly.